app/editor/src/features/content/form/components/upload/Upload.tsx

```diff
@@ -57,26 +57,44 @@ export const Upload: React.FC<IUploadProps> = ({
   stream,
   ...rest
 }) => {
+  const [loading, setLoading] = React.useState(true);
   const { isShowing, toggle } = useModal();
   const { values, setFieldValue } = useFormikContext<IContentForm>();
   const [, { ffmpeg }] = useWorkOrders();
   const { isShowing: showFFmpegModal, toggle: toggleFFmpeg } = useModal();
 
-  const [file, setFile] = React.useState<IFile>();
+  const [file, setFile] = React.useState<IFile | undefined>();
 
-  const fileName = generateName(file) ?? generateName(initFile);
+  const [fileName, setFileName] = React.useState<string | undefined>();
+  const [fileBlobUrl, setFileBlobUrl] = React.useState<string | undefined>();
   const fileReference = values.fileReferences.length ? values.fileReferences[0] : undefined;
   const processing = values.workOrders.some(
     (wo) =>
       wo.workType === WorkOrderTypeName.FFmpeg && wo.status === WorkOrderStatusName.InProgress,
   );
 
   React.useEffect(() => {
-    if (!!initFile) {
-      setFile(undefined);
+    if (!!initFile && loading) {
+      setFile(initFile);
+      setFileName(generateName(initFile));
+      setLoading(false);
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Always explain why you are not including all possible dependencies.  According to 'people' this means we're doing something wrong.  I however can't seem to figure out how to not cause infinite loops in some cases.

Reply:
In all my React dev experience this situation is unavoidable... I will try to see if there's a better way going forward. For now I'll put a note.